var/spack/repos/builtin/packages/upcxx/package.py

```diff
@@ -164,7 +164,11 @@ class Upcxx(Package, CudaPackage, ROCmPackage):
     depends_on("oneapi-level-zero@1.8.0:", when="+level_zero")
 
     # All flags should be passed to the build-env in autoconf-like vars
-    flag_handler = env_flags
+    def flag_handler(self, name, flags):
+        if name == "cxxflags":
+            if self.spec.satisfies("%oneapi@2025:"):
+                flags.append("-Wno-error=missing-template-arg-list-after-template-kw")
+        return (flags, None, None)
```

Comment:
@wspear @eugeneswalker IIUC this change merged in less than 4 hours without review by the package maintainer may have caused a regression.

The old behavior was `flag_handler = env_flags` which IIUC in the [documentation](https://spack.readthedocs.io/en/latest/packaging_guide.html#compiler-flags) returns `(None, flags, None)`. The new code introduced by this PR has effectively changed the behavior to match the `inject_flags` handler, which has probably never been tested with this package.

Was that change deliberate? Did you conduct extensive testing with non-default flags to validate the new behavior?

Reply:
@bonachea You are right, this was not intentional. To preserve the old behavior it is supposed to be `(None, flags, None)`. I am sorry for this mistake. I should have been more careful. What was tested was that UPCXX built in the CI environments here on Spack, and those tests passed, and that it fixed the OneAPI build error. Here is a reversion:
* https://github.com/spack/spack/pull/47512